Upgrade to Ubuntu 22.04 and update tparse

leblowl opened the pull request:
Hi,

I noticed the latest dev VM boxes (version 72 and 73) appear to be broken, due to an error in clang:
```
 runtime1: clang: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by clang)
    runtime1: clang: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by clang)
    runtime1: clang: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by clang)
    runtime1: clang: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by clang)
    runtime1: clang: /lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by clang)
    runtime1: clang: /lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by clang)
```
which I believe is due to the use of Ubuntu 20 in the dev boxes mismatching the environment clang is built in. I then found https://github.com/cilium/packer-ci-build/pull/358 and just applied a couple fixes to get it to build and satisfy review comments. I also noticed that the `tparse` installed in master doesn't include the `-progress` feature, so I updated that dependency. I tested the 4.19 and 5.4 images locally by running the cilium integration tests. Anyways, not sure if this is helpful or if ya'll are moving away from this infrastructure, but the docs still reference it https://docs.cilium.io/en/stable/contributing/development/dev_setup/ and perhaps this branch can unblock some folks who still want to use it.

pchaigno committed:
packer: Update to Ubuntu 22.04

Relates: https://github.com/cilium/cilium/pull/20943

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Tam Mach <tam.mach@cilium.io>

pchaigno committed:
packer: Update boot commands for Ubuntu 22.04

The old approach of passing the cloud-init config doesn't work anymore.
I found the new one from a random GitHub repository [1]. Yay GitHub!

1 - https://github.com/simula/nornet-vmimage-builder-scripts/blob/ce6490e94e040f3bf0304d9dd11da4abb2c1aa71/installer/ubuntu-22.04.json
Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
kernel: Install libssl1.1 for kernel downgrade

An older version of libssl is needed to install older kernels.

Co-authored-by: Tam Mach <tam.mach@cilium.io>
Signed-off-by: Paul Chaignon <paul@cilium.io>

sayboras committed:
packer: Add wait step for cloud-init completion

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Use forked netperf to fix compilation issue

We can just move it to cilium org if required :).

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Update cri-o version

This commit is to bump OS version and cri-o version to recent one.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
dep: Update packages for 22.04

This commit is to make sure additional package installation will not
fail due to not found error.

Signed-off-by: Tam Mach <tam.mach@cilium.com>

sayboras committed:
packer: Fix crash during cloud-config installation

```
/usr/bin/ckbcomp: Can not find file "symbols/en" in any known directory
```

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel/5.4: Update linux modules path for 5.4

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Update COMPRESS algo in initramfs.com

This is to avoid kernel panic due to the below issue during booting up.

```
VFS: Cannot open root device "UUID=..." or unknown-block(0,0): error -6
Please append a correct "root=" boot option; here are the available partitions:
Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
...
```

Suggested-by: Maxim Mikityanskiy <maxim@isovalent.com>
Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Add execute permission for kernel-next-tools.sh

Mainly for local development, as the CI is working as it was.

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
docker: Update the docker installation steps

This is to follow to the latest step mentioned in the official docs.

https://docs.docker.com/engine/install/ubuntu/

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
docker: Set alternatives for iptables

This is to avoid the below issue while starting up docker.

```
Mar 24 22:59:10 ubuntu-jammy dockerd[2820]: failed to start daemon: Error initializing network controller: error obtaining controller instance: failed to create NAT chain DOCKER: iptables failed: iptables -t nat -N DOCKER: iptables v1.8.7 (nf_tables): Could not fetch rule set generation id: Invalid argument
```

Relates: https://github.com/docker/for-linux/issues/1437

Signed-off-by: Tam Mach <tam.mach@cilium.io>

sayboras committed:
kernel: Disable cgroupv2 for 4.19

This is to support k8s-1.16-kernel-4.19 job.

leblowl committed:
Various feedback fixes

Signed-off-by: Lucas Leblow <lucasleblow@mailbox.org>

leblowl committed:
Upgrade tparse to include -progress feature

Signed-off-by: Lucas Leblow <lucasleblow@mailbox.org>

aanm commented:
/build

nbusseneau commented:
/build

nbusseneau commented:
/build

leblowl commented:
It looks like there was a temporary commit that enabled the post processors on PR jobs, which caused a conflict when uploading to Vagrant Cloud because the PR jobs and Master jobs have overlapping build IDs and upload the same artifact, using build ID as the version. Other than the post processors in the net-next build, everything built successfully. I reverted that commit and the post processors haven't changed in this PR, so I think it should be good now.

nbusseneau commented:
/build

leblowl commented:
This PR also removes the Jenkinsfile for the 4.9 build, so I think that check should be disabled. I don't think I have the ability to do that though

aanm commented:
@nbusseneau can we drop the build from Jenkins? I'll remove the "required" from the PR checks. I think we can merge this PR. Thank you for this @leblowl :clap: 

Next the image will be officially build and then we can update them in cilium/cilium.

aanm commented:
@leblowl I can't merge the PR. Do you mind rebasing against the latest tip of master?

nbusseneau commented:
I don't think we should drop 4.9, considering it's still used by older stable branches. We probably want to think a bit more about what we want to do going forward for the image maintenance considering only stable branches are still using the packer images.

leblowl commented:
Ok, any preference on dropping 4.9 for older stable branches? If you'd like to keep it around a bit longer, I can work on getting a 4.9 image building.

aanm commented:
@leblowl does the 4.9 need to be updated? I think we can leave it untouched and if we need it we can work on it afterwards.

leblowl commented:
@aanm I think since all the images share the same clang install script, when running clang it'll produce glibc errors like above. But since the 4.9 image itself is only used for older release builds (and those builds are using an older version of the image that works), like you are saying, we could just leave it and fix it if needed. I can restore the 4.9 files and rebase.

aanm commented:
that LGTM, @nbusseneau all good?

leblowl committed:
Disable cgroup v2 for 4.9 kernel

Signed-off-by: Lucas Leblow <lucasleblow@mailbox.org>

leblowl commented:
When I looked into it, the changes to upgrade the 4.9 image to Ubuntu 22.04 had already been done in previous commits. I just removed cgroup v2 support for the 4.9 image and it built successfully. I rebased on master. If you want to keep 4.9 on Ubuntu 20.04, I can do that too.

aanm commented:
/build

nbusseneau commented:
Yeah that's perfect thanks!
